### PR TITLE
feat(regex): Add a regex match operator for indexing

### LINKED ISSUE #30: Add a regex match operator for indexing
Add support for matching a regular expression in indexing, when only parts of a text value should be indexed, e.g. if text value is `by James Fox`, one could define indexed property:
```
value: {match("by (.*)")}
```
This could also work for arrayed values, e.g. for text value `Topics: Science, Nature, `
```
value: {match("(Topics: )? ([^,]+)")}
```
would return `[ 'Science','Nature' ]`

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #30

### CODE REVIEW

**Comment by dominique-pfister** [comment]
Note, that for testing, I fetch index definition and HTML page from the local file system (see `fileLoader` vs `repoLoader` in `html_json.js`). Usually, the index definition is loaded from:
```
https://raw.githubusercontent.com/${owner}/${repo}/${ref}/helix-index.yaml
```
but in testing, `owner` is `helix` and `repo` is `github.com--adobe--helix-index-pipelines`, and I have no clue how to build a git url to fetch the index file from that input.

**Comment by dominique-pfister** [comment]
@trieloff , I ended up parsing the index expression myself, it still abides to your new index config definition in https://github.com/adobe/helix-shared/pull/206, though

File: src/html_json.js [hunk 1]
```diff
@@ -12,61 +12,156 @@
...
+ * Fetch HTML page from a remote location, defined by owner, repo and path.
+ *
+ * @param {object} params
+ * @return HTML file
+ */
+async function fetchHTMLFromRepo(params) {
+  const {
+    index, owner, repo, path, logger,
+  } = params;
+  const pageUrl = index.fetch
```
Comment (trieloff) [inline src/html_json.js]: We should use a proper URI Template library here.

File: src/html_json.js [hunk 2]
```diff
@@ -12,61 +12,156 @@
...
+
 /**
  * Return a value in the DOM by evaluating an expression
  *
  * @param {HTMLElement} element
  * @param {string} expression
  */
 function getDOMValue(element, expression) {
-  const m = expression.match(/{([a-zA-Z]+)(\("([^"]+)"\))?}/);
+  const m = expression.match(/\${([a-zA-Z]+)\(('([^"]+)')?\)}/);
```
Comment (trieloff) [inline src/html_json.js]: Not sure how robust and generalizable this is, but I assume it's fine for theblog.

File: src/html_json.js [hunk 3]
```diff
@@ -112,26 +207,28 @@ module.exports.main = async (context, action) => {
     owner, repo, ref, path,
   } = action.request.params;
 
-  // TODO: this should be loaded once or even passed in context
-  const config = await loadConfig(owner, repo, ref);
+  // Use a file loader if this is a local setup
+  const loader = (owner === 'helix') ? fileLoader : repoLoader;
```
Comment (tripodsan) [inline src/html_json.js]: - how does the `owner===helix` indicate a local setup?
- why using a local setup at all? if it's just for development, I would use a `develop` flag.

Comment (dominique-pfister) [reply src/html_json.js]: local setup meaning it was started by `helix-simulator`, in `HelixProject.emulateGit`

**Pull request merged**